wandb/sdk/wandb_metadata.py

```diff
@@ -16,9 +20,18 @@
     from typing_extensions import Self
 
 
+is_pydantic_v2 = int(PYDANTIC_VERSION[0]) == 2
+
+if is_pydantic_v2:
+    from pydantic import model_validator
+
+else:
+    pass
```

Comment:
what happens in pydantic 1?

Reply:
metadata becomes read-only from the user perspective if they are on v1.